### ISSUE TITLE
Feedback 1

### ISSUE THREAD

**Issue by djdupre** [avidant/test-bayes#58]
Here is some general feedback for this assignment: https://courses.cs.washington.edu/courses/cse403/18sp/projects/proj08b-address-feedback.html

I decided to review Test Bayes by first trying it out on a project to reveal any problems with usability. Then I looked at pull requests and commits to examine process and briefly checked the source code for style.

### Usage

Copying your projects jar around is bad for maintainability. Your README says to just copy the jar into my project's root folder, but it skips the step of actually adding the jar as a dependency. Test Bayes would be more usable and maintainable if it were uploaded to the Maven Central Repository so it could be automatically downloaded and kept up-to-date.

Maintaining a list of all tests in the `@Reorder.SmartOrder({...})` annotation seems like it would be annoying for large projects. Could there be a way to automatically reorder all tests without a need for this list?

When I tried to run my tests with Test Bayes, I got a `java.lang.Exception: No runnable methods` error. See #57 for details.

### Process

It's great that you're using pull requests, but changes to the code are hard to follow when reports are mixed in. You could fix this by labeling all reports with a `report` tag so they can be filtered out.

It's concerning that these are the first issues for your project. GitHub's issue tracker is a great way to track bugs and features (though it will soon be surpassed by [Track Forever](https://cse403trackforever.firebaseapp.com)).

It would also be nice to see code coverage with JaCoCo, for example, with coveralls and travis-ci to view the coverage online.

### Code

- You should be using `Double.compare` in `Probability.java`
- There seems to be a lot of repeated code in `Bayes.java`, especially between `buildPasscond` and `buildFailcond`. You should consider refactoring these functions.
- The exceptions in the `IndividualClassRunner` constructor can be replaced with `IllegalAccessException|InstantiationException e`.

**Comment by avidant** [comment]
**Usage**
- Copying the jar is only temporary. We do not want to add this to the Maven Central Repository while this is still in development. Once the project is at a stage where are happy to share the same, we will be doing so
- I did not realize that we had missed that step. Will fix that in #59
- The Annotation is also a temporary measure. Our final solution would not require this

**Process**
- That's a great idea! We will tag all our reports for sure
- We'll start using issues. We have been doing so internally on Slack, but will start using GitHub's issue tracker
- We can certainly add something to see coverage. Do you think something like the Travis CI build status on our README would be good? #60

**Code**
- That is something we missed. We will fix that right away! #61
- `Bayes.java` is currently being refactored. So we will be sure to fix the redundancy #62
- That's an interesting solution I didn't know about. Will fix that too. #63

I am closing this as I think the other issues I have created should cover the issues you brought up. But please do reopen if you think there's something I missed!

**Comment by djdupre** [comment]
I think adding the Travis CI build status is a great idea. If you decide to use coveralls, you can add a badge for coverage too.

By the way, most of that code feedback was picked up by IntelliJ IDEA. It's free for students and can save you a lot of time finding stuff like this.
